### PR TITLE
ray.objects() should only return objects created by local driver

### LINKED ISSUE #7904: ray.get hangs when it is called on an objectID that is not in the local storage, but in the plasma store.
### What is the problem?
Originally from this stackoverflow question. https://stackoverflow.com/questions/60975923/storing-and-retrieving-object-in-ray-io

When a driver tries to call ray.get on an objectID that is not created by itself, it hangs forever.

*Ray version and other system information (Python version, TensorFlow version, OS):*
Python 3.7
Latest wheel

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):
Driver 1
```python3
import ray
ray.init()
ray.put(1)
```

Driver 2
```python3
import ray
ray.init(address='auto')
ray.get(list(ray.objects()))
```

Driver2 hangs until `ray.get` is timed out. 

Driver 2 cannot get an object in the plasma store when the object is not stored in its local storage because the object was created by a different driver. In this case, it should return an error or look for plasma store, but it just hangs. 

If we cannot run your script, we cannot fix your issue.

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://ray.readthedocs.io/en/latest/installation.html).

### PR DESCRIPTION
**Pull request by walterddr**
## Why are these changes needed?

Return only the objects created by local driver since ray.objects() are not meant to communicate with plasma store.

## Related issue number

Closes: #7904 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Already exist unit tests

### CODE REVIEW

File: python/ray/state.py [hunk 1]
```diff
@@ -288,6 +290,9 @@ def object_table(self, object_id=None):
             # Return information about a single object ID.
             return self._object_table(object_id)
         else:
+            # Find the local driver / job
+            global_worker = ray.WorkerID(ray.worker.global_worker.worker_id)
```
Comment (rkooo567) [inline python/ray/state.py:294]: Isn't it supposed to be `global_worker_id`?

Comment (walterddr) [reply python/ray/state.py:294]: hmm. I am not sure I found that . are you referring to `ray.worker.global_worker_id` ?

Comment (rkooo567) [reply python/ray/state.py:294]: oh what I meant was the variable name. 
```suggestion
            global_worker_id = ray.WorkerID(ray.worker.global_worker.worker_id)
```

**Review by rkooo567 — COMMENTED** [review]
Test seems to fail. I will try re-running them. How did you test this functionality?

File: python/ray/tests/test_advanced_3.py [hunk 2]
```diff
@@ -128,6 +128,9 @@ def test_global_state_api(shutdown_only):
 
     assert ray.objects() == {}
 
+    ray.put(1)
```
Comment (walterddr) [inline python/ray/tests/test_advanced_3.py:131]: @rkooo567 I added a test here just to make sure it could acquire items back

**Review by walterddr — COMMENTED** [review]
addressed diff comment.

Not sure why tests are failing, I manually ran all tests with `ray.objects()` in it and they all look fine. FYI most of the tests I found is in `test_advanced_3.py`

File: python/ray/tests/test_advanced_3.py (hunk 2 above)
Comment (rkooo567) [reply python/ray/tests/test_advanced_3.py:131]: That's awesome!

**Comment by rkooo567** [comment]
@walterddr Sorry for getting back late! I was a little busy with other work. This PR look very close to the end state. @ericl Let me know your thoughts about this API changes. 

Also, when we change the public facing API behaviors, we are required to have some discussion. I will bring up this API changes to other contributors.

**Comment by walterddr** [comment]
hmm. based on the discussion originally. I dont think this `ray.objects()` are meant to be used inside workers. 
I can add some checks to make sure `ray.objects()` throws exception when used within a worker. is that something that can address your concern?

**Comment by walterddr** [comment]
Hmm. there seems to be some new changes in #8298 that makes the exception different.
previously it was crashing the core but now it actually causes a stuck kernel. I will try to rebase and change it a bit.

**Pull request closed without merging**